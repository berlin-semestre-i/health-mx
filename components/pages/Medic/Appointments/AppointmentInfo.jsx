import React from 'react'
import styled from 'styled-components'
import media from 'styled-media-query'

const AppointmentInfo = ({ children }) => (
  <ConsultationInfo>
    { children }
  </ConsultationInfo>
)

export default AppointmentInfo

const ConsultationInfo = styled.div`
  font-size: 12px;
  color: #3d5170;
  font-weight: 400;

  ${media.greaterThan('large')`
    width: 85%;
    display: flex;
    justify-content: space-between;
  `} 
`