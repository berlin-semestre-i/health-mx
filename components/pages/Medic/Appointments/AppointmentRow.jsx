import React from 'react'
import { Grid, Icon, Header } from 'semantic-ui-react'
import Avatar from '../../../shared/Avatar'
import ConsultationInfo from './AppointmentInfo'
import ValueItem from '../../../shared/ValueItem'
import styled from 'styled-components'
import Link from 'next/link'
import media from 'styled-media-query'

const AppointmentRow = ({ appointment, className }) => (
  <React.Fragment>
    <Link href="/medic/medicalReport">
      <a>
        <ConsultationRow>
          <AvatarContainer mobile={16} tablet={3} computer={1}>
            <Avatar src={appointment.profile}/>
          </AvatarContainer>
          <ConsultationInfoContainer mobile={16} tablet={13} computer={10}>
            <BeneficiaryName>
              { appointment.name }
            </BeneficiaryName>
            <ConsultationInfo>
              <ValueItem keyName="Edad" value={appointment.age} />
              <ValueItem keyName="Estado" value={appointment.status} />
              <ValueItem keyName="Tratamiento" value={appointment.treatment} />
              <ValueItem keyName="Ãšltima cita" value={appointment.previous} />
            </ConsultationInfo>
          </ConsultationInfoContainer>
          <ConsultationTime computer={4} tablet={4} mobile={16} className={className}>
            <Header as="h3">Hora</Header>
            <Header as="h1">{ appointment.time }</Header>
          </ConsultationTime>
          <ChevronContainer computer={1}>
            <Icon name="chevron right" />
          </ChevronContainer>
        </ConsultationRow>
      </a>
    </Link>
  </React.Fragment>
)

export default AppointmentRow

const BeneficiaryName = styled.span`
  font-weight: 600;
`
const ConsultationRow = styled(Grid)`
  >.column:not(.row):last-child {
    padding-top: 2rem;
  }

  ${media.greaterThan('large')`
    >.column:not(.row):nth-child(2) {
      padding-top: 1.5rem;
    }
  `} 
`
const ConsultationTime = styled(Grid.Column)`
  text-align: right;

  >.ui.header {
    margin: 0;
  }

  >h3.ui.header {
    font-size: 13px;
  }

  >h1.ui.header {
    font-size: 26px;
    font-weight: 500;
    color: #808AA2;
  }

  &.is-active>h1.ui.header {
    color: #00B5CE;
  }
`
const ChevronContainer = styled(Grid.Column)`
  >.icon {
    font-size: 20px;
    font-weight: 700;
    color: #808AA2;
  }

  ${media.lessThan('small')`
    &&.column:not(.row) {
      display: none;
    }
  `} 
`
const AvatarContainer = styled(Grid.Column)`
  text-align: center;

  ${media.between('medium','1025px')`
    [class*="one wide computer"]&&&.column {
      width: 12%!important;
    }
  `}

  ${media.lessThan('medium')`
    &&&.column:not(.row) {
      padding-bottom: 0;
    }
  `}
`
const ConsultationInfoContainer = styled(Grid.Column)`
  ${media.between('medium','1025px')`
    [class*="ten wide computer"]&&&.column {
      width: 55%!important;
    }
  `}

  >a:not(.ui) {
    font-weight: 600;
  }
`