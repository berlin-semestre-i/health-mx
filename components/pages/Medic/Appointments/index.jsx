import React, { PureComponent } from 'react'
import PageHeader from '../../../shared/PageHeader'
import AppointmentRow from './AppointmentRow'
import Card from '../../../shared/Card'
import { Grid, Header } from 'semantic-ui-react'
import styled from 'styled-components'

class Appointments extends PureComponent {
  state = {
    appointment: {
      name: 'María del Carmen Hinojosa Ramírez',
      date: '20 Nov, 2018',
      time: '08:30 am',
      treatment: 'N/A',
      status: 'Regular',
      age: '25',
      previous: '12 Jul, 2018',
      profile: '../../../static/images/woman-avatar.png',
    },
    appointment1: {
      name: 'José Armando Domínguez Pérez',
      date: '20 Nov, 2018',
      time: '10:00 am',
      status: 'Regular',
      age: '30',
      treatment: 'En curso',
      previous: '22 Sept, 2018',
      profile: '../../../static/images/man-avatar.png',
    },
  }

  render() {
    const { appointment, appointment1 } = this.state

    return (
      <React.Fragment>
        <Grid>
          <PageHeader
            title="Citas Médicas"
            subtitle="Citas Pendientes"
          />
          <Card header="Próxima Cita">
            <AppointmentRow className="is-active" appointment={appointment}/>
          </Card>
          <Grid.Row>
            <AppointmentContainer>
              <AppointmentRow appointment={appointment1}/>
            </AppointmentContainer>
          </Grid.Row>
          <Header as="h1">
            Citas Completadas
          </Header>
          <Grid.Row>
            <AppointmentContainer>
              <AppointmentRow appointment={appointment1}/>
            </AppointmentContainer>
          </Grid.Row>
        </Grid>
      </React.Fragment>
    )
  }
}

const AppointmentContainer = styled(Grid.Column)`
  padding-top: 1rem;

  &&&.column {
    padding-left: 1.75rem;
    padding-right: 1.75rem;
  }
`

export default Appointments