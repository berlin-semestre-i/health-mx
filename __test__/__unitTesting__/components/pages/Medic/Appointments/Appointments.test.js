import React from 'react'
import { shallow, mount } from 'enzyme'
import Appointments from '../../../../../../components/pages/Medic/Appointments/index'

describe('Appointments', () => {
  it('renders', () => {
    const component = shallow(<Appointments />)
    expect(component).toMatchSnapshot()
  })

  it('has one card', () => {
    const wrapper = mount(<Appointments />)
    expect(wrapper.find('div.card').length).toBe(1)
    wrapper.unmount()
  })
})