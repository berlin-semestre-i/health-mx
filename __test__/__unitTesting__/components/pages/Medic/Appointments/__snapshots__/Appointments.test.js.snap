// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointments renders 1`] = `
<Fragment>
  <Grid>
    <PageHeader
      subtitle="Citas Pendientes"
      title="Citas Médicas"
    />
    <CustomCard
      header="Próxima Cita"
    >
      <AppointmentRow
        appointment={
          Object {
            "age": "25",
            "date": "20 Nov, 2018",
            "name": "María del Carmen Hinojosa Ramírez",
            "previous": "12 Jul, 2018",
            "profile": "../../../static/images/woman-avatar.png",
            "status": "Regular",
            "time": "08:30 am",
            "treatment": "N/A",
          }
        }
        className="is-active"
      />
    </CustomCard>
    <GridRow>
      <Appointments__AppointmentContainer>
        <AppointmentRow
          appointment={
            Object {
              "age": "30",
              "date": "20 Nov, 2018",
              "name": "José Armando Domínguez Pérez",
              "previous": "22 Sept, 2018",
              "profile": "../../../static/images/man-avatar.png",
              "status": "Regular",
              "time": "10:00 am",
              "treatment": "En curso",
            }
          }
        />
      </Appointments__AppointmentContainer>
    </GridRow>
    <Header
      as="h1"
    >
      Citas Completadas
    </Header>
    <GridRow>
      <Appointments__AppointmentContainer>
        <AppointmentRow
          appointment={
            Object {
              "age": "30",
              "date": "20 Nov, 2018",
              "name": "José Armando Domínguez Pérez",
              "previous": "22 Sept, 2018",
              "profile": "../../../static/images/man-avatar.png",
              "status": "Regular",
              "time": "10:00 am",
              "treatment": "En curso",
            }
          }
        />
      </Appointments__AppointmentContainer>
    </GridRow>
  </Grid>
</Fragment>
`;
